pkg/osquery/table/chrome_login_keychain.go

```diff
@@ -65,41 +71,65 @@ func (c *ChromeLoginKeychain) generateForPath(ctx context.Context, path string)
 
 	// loop through all the sqlite rows and add them as osquery rows in the results map
 	for rows.Next() { // we initialize these variables for every row, that way we don't have data from the previous iteration
-		var origin_url string
-		var action_url string
 		var username_value string
-		if err := rows.Scan(&origin_url, &action_url, &username_value); err != nil {
+		var username_freq string
+		if err := rows.Scan(&username_value, &username_freq); err != nil {
 			return nil, errors.Wrap(err, "scanning chrome login keychain db row")
 		}
-
 		results = append(results, map[string]string{
-			"origin_url":     origin_url,
-			"action_url":     action_url,
+			"username":       username,
 			"username_value": username_value,
+			"frequency":      username_freq,
 		})
 	}
 	return results, nil
 }
 
 func (c *ChromeLoginKeychain) generate(ctx context.Context, queryContext table.QueryContext) ([]map[string]string, error) {
-	paths, err := findFileInUserDirs("Library/Application Support/Google/Chrome/Default/Login Data")
-	if err != nil {
-		return nil, errors.Wrap(err, "find chrome login data sqlite DBs")
+	var usernames []string
+	q, ok := queryContext.Constraints["username"]
+	if ok && len(q.Constraints) != 0 {
+		for _, constraint := range q.Constraints {
+			usernames = append(usernames, constraint.Expression)
+		}
+	} else {
+		return nil, errors.New("The kolide_chrome_login_keychain table requires that you specify a constraint WHERE username =")
 	}
 
 	var results []map[string]string
-	for _, path := range paths {
-		res, err := c.generateForPath(ctx, path)
-		if err != nil {
-			level.Info(c.logger).Log(
-				"msg", "Generating chrome keychain result",
-				"path", path,
-				"err", err,
-			)
-			continue
+	osProfileDirs, ok := profileDirs[runtime.GOOS]
+	if !ok {
+		osProfileDirs, ok = profileDirs["default"]
+		if !ok {
+			return results, errors.New("No profileDir for this platform")
 		}
-		results = append(results, res...)
 	}
 
+	for _, username := range usernames {
+		for _, profileDir := range osProfileDirs {
+			userPaths, err := findFileInUserDirs(filepath.Join(profileDir, "Default/Login Data"), WithUsername(username))
```

Comment:
What is this `filepath.Join`?

Reply:
`Default/Login Data` is the sqlite db.

Reply:
Are you asking why filepath.Join vs path.Join?

Reply:
Oh, that's the sqlite db? The space in the filename made me suspect it was darwin specific. If that's correct on all platforms, it fine. Else it should get dropped into the constants at the top